accounts/usbwallet/ledger.go

```diff
@@ -325,11 +338,27 @@ func (w *ledgerDriver) ledgerSign(derivationPath []uint32, tx *types.Transaction
 	for i, component := range derivationPath {
 		binary.BigEndian.PutUint32(path[1+4*i:], component)
 	}
-	// Create the transaction RLP based on whether legacy or EIP155 signing was requested
+
 	var (
 		txrlp []byte
 		err   error
 	)
+
+	if ledger.IsERC20Transfer(tx.Data()) {
+		err = w.ledgerProvideERC20(*tx.To(), chainID)
+		if err != nil && err != ledger.ErrCouldNotFindToken {
+			return common.Address{}, nil, err
+		}
+
+		if tx.FeeCurrency() != nil {
```

Comment:
I don't know if I'm following the case here...
If the feeCurrency is for example cUSD, it means that the transaction will be a cUSD one? It's not possible to just transfer cGOLD paying fees using a stable token?

Reply:
Yeah, it should be possible!

There are two calls to `ledgerProvideERC20` - one for the `To` and one for the `FeeCurrency`. The Ledger app knows to save the last 2 calls of `ledgerProvideERC20` it received, so it should have info about both.

Reply:
Ahhh but does means that you are somehow deleting the `To` field from the transaction? otherwise you'll getting both times the same contractData